spec/features/orders/user_cancel_spec.rb

```diff
@@ -0,0 +1,150 @@
+
+    @pull_toy = @dog_shop.items.create(name: "Pully Toy", description: "Great pull toy!", price: 10, image: "http://lovencaretoys.com/image/cache/dog/tug-toy-dog-pull-9010_2-800x800.jpg", inventory: 32)
+    @dog_bone = @dog_shop.items.create(name: "Doggy Bone", description: "They'll love it!", price: 21, image: "https://img.chewy.com/is/image/catalog/54226_MAIN._AC_SL1500_V1534449573_.jpg", active?:false, inventory: 21)
+
+    @merchant_user1 = @bike_shop.users.create!(name: "Johnny",
+                                street_address: "123 Jonny Way",
+                                city: "Johnsonville",
+                                state: 'TN',
+                                zip_code: 12345,
+                                email: "roman4444@example.com",
+                                password: "hamburger042",
+                                role: 1
+                              )
+
+    @merchant_user2 = @dog_shop.users.create!(name: "Jeremiah",
+                                street_address: "7777 Rastafari Way",
+                                city: "Jamaica",
+                                state: 'FL',
+                                zip_code: 46766,
+                                email: "ziggymarley@example.com",
+                                password: "irie333",
+                                role: 1
+                              )
+
+    @walter = User.create!(name: "Walter White",
+                                 street_address: "6230 Bluerock Ln",
+                                 city: "Albuquerque",
+                                 state: 'NM',
+                                 zip_code: 44565,
+                                 email: "heisenberg@example.com",
+                                 password: "method3",
+                                 role: 0
+                                  )
+  end
+  it "all merchants can fulfill items on an order" do
+    visit '/login'
+
+    fill_in 'email', with: @walter.email
+    fill_in 'password', with: @walter.password
+
+    click_on 'Log In'
+
+    click_on("All Items")
+
+    click_on("GatorSkins")
+    click_on("Add To Cart")
+
+    click_on("Doggy Bone")
+    click_on("Add To Cart")
+
+    click_on("Pully Toy")
+    click_on("Add To Cart")
+
+    click_on("Pully Toy")
+    click_on("Add To Cart")
+
+    visit "/cart"
+
+    click_on("Checkout")
+
+    fill_in :name, with: @walter.name
+    fill_in :address, with: @walter.street_address
+    fill_in :city, with: @walter.city
+    fill_in :state, with: @walter.state
+    fill_in :zip, with: @walter.zip_code
+
+    click_button("Create Order")
+
+    order = Order.last
+
+    visit '/profile/orders'
+
+    click_on(order.id)
+    click_on("Cancel Order")
+
+    expect(current_path).to eq("/profile")
+
+    expect(page).to have_content("Your order has been cancelled.")
+
+    click_on("My Orders")
+
+    expect(page).to have_content("Current status: cancelled")
```

Comment:
we could make this test more robust by testing that this content is within a `div` for that specific `order`